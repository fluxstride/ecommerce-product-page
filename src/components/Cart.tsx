import { memo, MutableRefObject, useRef } from 'react';
import cartIcon from '../assets/images/icon-cart.svg';
import useOnClickOutside from '../hooks/useOnClickOutside';
import useShow from '../hooks/useShow';

const Cart = () => {
  const {
    show: cartVisible,
    hideElement: hideCart,
    toggleShow: toggleCartShow,
  } = useShow();
  const cartRef = useRef<Element | null>(null);
  useOnClickOutside(cartRef, hideCart);

  return (
    <div
      ref={cartRef as MutableRefObject<HTMLDivElement>}
      className='cart_container'
    >
      <button type='button' className='cart_button' onClick={toggleCartShow}>
        <img src={cartIcon} alt='cart' />
      </button>
      {cartVisible && (
        <div className='cart'>
          <div className='inner sec_pad'>
            <h3>Cart</h3>
            <hr />
            <button type='button' onClick={toggleCartShow}>
              checkout
            </button>
          </div>
        </div>
      )}
    </div>
  );
};

export default memo(Cart);
