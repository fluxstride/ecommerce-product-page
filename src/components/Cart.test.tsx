import React from 'react';
import render from 'react-test-renderer';
import Cart from './Cart';

describe('<Cart/>', () => {
  it('expect the Cart component to match the given snapshot', () => {
    const component = render.create(<Cart />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
