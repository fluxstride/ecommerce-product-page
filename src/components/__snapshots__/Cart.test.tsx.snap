// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart/> expect the Cart component to match the given snapshot 1`] = `
<div
  className="cart_container"
>
  <button
    className="cart_button"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="cart"
      src="icon-cart.svg"
    />
  </button>
</div>
`;
